sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3343,6 +3344,296 @@ object functions {
+    ArrayTransform(column.expr, createLambda(f))
+  }
+
+  /**
+   * (Java-specific) Returns whether a predicate holds for one or more elements in the array.
+   *
+   * @group collection_funcs
+   */
+  def exists(column: Column, f: JavaFunction[Column, Column]): Column = withExpr {
+    ArrayExists(column.expr, createLambda(f))
+  }
+
+  /**
+   * (Java-specific) Returns whether a predicate holds for every element in the array.
+   *
+   * @group collection_funcs
+   */
+  def forall(column: Column, f: JavaFunction[Column, Column]): Column = withExpr {
+    ArrayForAll(column.expr, createLambda(f))
+  }
+
+  /**
+   * (Java-specific) Returns an array of elements for which a predicate holds in a given array.
+   *
+   * @group collection_funcs
+   */
+  def filter(column: Column, f: JavaFunction[Column, Column]): Column = withExpr {
+    ArrayFilter(column.expr, createLambda(f))
+  }
+
+  /**
+   * (Java-specific) Applies a binary operator to an initial state and all elements in the array,
+   * and reduces this to a single state. The final state is converted into the final result
+   * by applying a finish function.
+   *
+   * @group collection_funcs
+   */
+  def aggregate(expr: Column, zero: Column, merge: JavaFunction2[Column, Column, Column],
+                finish: JavaFunction[Column, Column]): Column = withExpr {
+    ArrayAggregate(
+      expr.expr,
+      zero.expr,
+      createLambda(merge),
+      createLambda(finish)
+    )
+  }
+
+  /**
+   * (Java-specific) Applies a binary operator to an initial state and all elements in the array,
+   * and reduces this to a single state.
+   *
+   * @group collection_funcs
+   */
+  def aggregate(expr: Column, zero: Column, merge: JavaFunction2[Column, Column, Column]): Column =
+    aggregate(
+      expr, zero, merge, new JavaFunction[Column, Column] { def call(c: Column): Column = c })
+
+  /**
+   * (Java-specific) Merge two given arrays, element-wise, into a signle array using a function.
+   * If one array is shorter, nulls are appended at the end to match the length of the longer
+   * array, before applying the function.
+   *
+   * @group collection_funcs
+   */
+  def zip_with(left: Column, right: Column, f: JavaFunction2[Column, Column, Column]): Column =
+    withExpr {
+      ZipWith(left.expr, right.expr, createLambda(f))
+    }
+
+  /**
+   * (Java-specific) Applies a function to every key-value pair in a map and returns
+   * a map with the results of those applications as the new keys for the pairs.
+   *
+   * @group collection_funcs
+   */
+  def transform_keys(expr: Column, f: JavaFunction2[Column, Column, Column]): Column = withExpr {
+    TransformKeys(expr.expr, createLambda(f))
+  }
+
+  /**
+   * (Java-specific) Applies a function to every key-value pair in a map and returns
+   * a map with the results of those applications as the new values for the pairs.
+   *
+   * @group collection_funcs
```

Comment:
Shall we add `@since 3.0.0`?

Reply:
https://github.com/apache/spark/pull/24232/commits/96fb0ad110fc0e0cc3e8f8097fd370143999a7c2